survival/utils.py

```diff
@@ -0,0 +1,31 @@
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+def intersect_series_with_index(index: pd.Index, data: pd.Series) -> pd.Series:
+    """
+    returns series intersected with given index
+    """
+    aligned_series = data.loc[index.intersection(data.index)]
+    return aligned_series
+
+def intersect_df(dataframes=()):
```

Comment:
not clear name - maybe get_dataframes_with_intersecting_indecies

no need give empty tuple this code will raise index error with this default value

add typehint for input and ouput: `List[Union[pd.Series, pd.DataFrame]]`